Framework/WorkflowAlgorithms/src/EQSANSLoad.cpp

```diff
@@ -343,12 +332,13 @@ void EQSANSLoad::moveToBeamCenter() {
       dataWS->getInstrument()->getNumberParameter("number-of-x-pixels")[0]);
   int ny_pixels = static_cast<int>(
       dataWS->getInstrument()->getNumberParameter("number-of-y-pixels")[0]);
-  V3D pixel_first = dataWS->getInstrument()->getDetector(0)->getPos();
+  const auto &detInfo = dataWS->detectorInfo();
+  const V3D pixel_first = detInfo.position(0);
   int detIDx = EQSANSInstrument::getDetectorFromPixel(nx_pixels - 1, 0, dataWS);
   int detIDy = EQSANSInstrument::getDetectorFromPixel(0, ny_pixels - 1, dataWS);
 
-  V3D pixel_last_x = dataWS->getInstrument()->getDetector(detIDx)->getPos();
-  V3D pixel_last_y = dataWS->getInstrument()->getDetector(detIDy)->getPos();
+  const V3D pixel_last_x = detInfo.position(detIDx);
+  const V3D pixel_last_y = detInfo.position(detIDy);
```

Comment:
Same here. Need to convert ID to index: `detInfo.position(detInfo.indexOf(detIDx))`.